src/py21cmfast/wrapper.py

```diff
@@ -1884,11 +1884,31 @@ def run_coeval(
         else:
             redshifts = [min(redshift)]
 
+        # USE_MINI_HALOS needs all perturb_field in redshifts!
+        if flag_options.USE_MINI_HALOS:
+            perturb_all_scroll = []
+            for ii in range(len(redshift)):
+                perturb_all_scroll += [perturb[ii]]
+            for z in redshifts:
+                perturb_all_scroll += [
```

Comment:
I don't think that generating another list of perturb_field boxes is the way to go as it can require a large amount of memory. Given that its something that is only required with `USE_MINI_HALOS`, perhaps my suggestion is better as it only requires one box at a time to be kept around. Theoretically, you could add another box (like st, ib) which can be populated with the previous perturb field as it goes through the loop (i.e. it starts at None, then after each iteration it is populated with the perturb field used at the previous redshift snapshot).